### PR TITLE
Allow ContentResponse to receive multiple components next to a title

### LINKED ISSUE #2302: Page layout allows for buttons in-line with page title
**Describe the problem/challenge you have**
Currently, the page layout doesn't allow for a button inline with the title. Adding buttons creates a row that adds page height and is misaligned.

**Describe the solution you'd like**
We would like to have some action buttons right-aligned, inline with the title.

Example: Screen Shot 2021-04-12 at 10 37 21

### PR DESCRIPTION
**Pull request by ftovaro**
Signed-off-by: ftovaro <f.tovar12@gmail.com>

**What this PR does / why we need it**:
Allows ContentResponse to receive multiple components, this is an extension our titles and gives the option to have complex titles with buttons, links, icons and other components.

image

**Which issue(s) this PR fixes**
- Fixes #2302 

**Special notes for your reviewer**:

**Release note**:
```
release-note
```

### CODE REVIEW

File: pkg/view/component/component.go [hunk 1]
```diff
@@ -34,6 +37,15 @@ func NewContentResponse(title []TitleComponent) *ContentResponse {
 	}
 }
 
+// AddTitleComponents Adds any number of components to accompany the title
+func (c *ContentResponse) AddTitleComponents(components ...Component) {
+	for i := range components {
```
Comment (lenriquez) [inline pkg/view/component/component.go:42]: Why not just assign components to TitleComponents

File: web/src/app/modules/shared/components/presentation/breadcrumb/breadcrumb.component.html [hunk 2]
```diff
@@ -8,7 +8,15 @@
...
-  <h2 class="breadcrumb-header">
-    {{ header }}
-  </h2>
+  <div class="title-container">
+    <h2 class="breadcrumb-header">
+      <span>{{ header }}</span>
+    </h2>
+    <div
+      *ngFor="let component of titleComponents; let i = index; trackBy: identifyPath"
+      class=""
```
Comment (lenriquez) [inline web/src/app/modules/shared/components/presentation/breadcrumb/breadcrumb.component.html:17]: I believe the class="" is not necessary

File: pkg/view/component/component.go (hunk 1 above)
Comment (ftovaro) [reply pkg/view/component/component.go:42]: To check first if we are receiving nil components

File: web/src/app/modules/shared/components/presentation/tabs/tabs.component.html [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
     <app-breadcrumb [path]="title"> </app-breadcrumb>
   </div>
-  <div class="clr-col-2">
+  <div class="clr-col-6">
+    <div *ngIf="titleComponents?.length > 1" class="title-container">
+      <div *ngFor="let component of titleComponents; let i = index; trackBy: identifyPath" class="item">
+        <app-view-container [view]="component"></app-view-container>
+      </div>
+    </div>
     <div *ngIf="buttonGroup" class="action-buttons">
```
Comment (GuessWhoSamFoo) [inline web/src/app/modules/shared/components/presentation/tabs/tabs.component.html:11]: This button group can be removed in favor of using title components altogether. `AddButton` in `component.go` will need to be updated to add a button into title components and buttonGroup can be removed.

**Review by GuessWhoSamFoo — CHANGES_REQUESTED** [review]

File: web/src/app/modules/shared/components/presentation/tabs/tabs.component.html [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
-  <div class="clr-col-10">
+  <div class="clr-col-6">
     <app-breadcrumb [path]="title"> </app-breadcrumb>
   </div>
-  <div class="clr-col-2">
-    <div *ngIf="buttonGroup" class="action-buttons">
-      <app-button-group [view]="buttonGroup"></app-button-group>
+  <div class="clr-col-6">
+    <div *ngIf="titleComponents?.length > 0" class="title-container">
+      <div *ngFor="let component of titleComponents; let i = index; trackBy: identifyPath" class="item">
```
Comment (GuessWhoSamFoo) [inline web/src/app/modules/shared/components/presentation/tabs/tabs.component.html:7]: `identifyPath` is a unique trackbyFn for breadcrumbs.

Remove `let i = index` and change the trackByFn to `trackByIndex` via `import trackByIndex from 'src/app/util/trackBy/trackByIndex';`. Without this set correctly, components passed through this method will update with each new content response which can be problematic for any persisting UI states (e.g. opened modals)

File: web/src/app/modules/sugarloaf/components/smart/content/content.component.html [hunk 5]
```diff
@@ -9,6 +9,7 @@
           [buttonGroup]="buttonGroup"
```
Comment (GuessWhoSamFoo) [inline web/src/app/modules/sugarloaf/components/smart/content/content.component.html:9]: This can go away now as well.

**Review by GuessWhoSamFoo — CHANGES_REQUESTED** [review]

**Review by GuessWhoSamFoo — APPROVED** [review]

**Comment by GuessWhoSamFoo** [comment]
Great job!

**Pull request merged**